lib/TPP/ConvertXsmmToFunc.cpp

```diff
@@ -339,12 +339,36 @@ static func::CallOp buildDispatchCall(Location loc,
   return call;
 }
 
-// TODO: We can derive all the dispatch from the same base class and use
-// this method. Left for future work as require changes on the td.
+template <typename OpTy,
+          typename = std::enable_if_t<
+              std::is_same<OpTy, xsmm::UnaryDispatchOp>::value ||
+              std::is_same<OpTy, xsmm::BinaryDispatchOp>::value ||
+              std::is_same<OpTy, xsmm::TernaryDispatchOp>::value>>
+void addKindOperand(RewriterBase &rewriter, OpTy dispatchOp,
+                    SmallVectorImpl<Value> &dispatchOperands,
+                    SmallVectorImpl<Type> &dispatchOperandTypes) {
+  Location loc = dispatchOp.getLoc();
+  IntegerType integer64 = IntegerType::get(rewriter.getContext(), 64);
+  dispatchOperands.push_back(rewriter.create<arith::ConstantOp>(
+      loc, integer64, dispatchOp.getKindAttr()));
+  dispatchOperandTypes.push_back(integer64);
+}
+
+void addKindOperand(RewriterBase &rewriter, MatmulDispatchOp dispatchOp,
+                    SmallVectorImpl<Value> &dispatchOperands,
+                    SmallVectorImpl<Type> &dispatchOperandTypes) {
+  /* do nothing */
+}
+
+void addKindOperand(RewriterBase &rewriter, BrgemmDispatchOp dispatchOp,
+                    SmallVectorImpl<Value> &dispatchOperands,
+                    SmallVectorImpl<Type> &dispatchOperandTypes) {
+  /* do nothing */
+}
```

Comment:
Are these needed? There's an if type check before calling this function anyway

Reply:
Yes, needed. Otherwise the compiler complains. Need to see specialization for GEMM and BRGEMM